### PR TITLE
#429 minimap centering based on camera position

### LINKED ISSUE #429: [BUG] Camera Issue - Minimap centering based off of camera position
**Describe the bug**
The minimap is currently centering itself based off of the position of the player's camera when it should be centered based on the player character's location. This is especially noticeable when playing with a very high camera distance or when playing as a titan as they inherently have a higher camera distance. If you set camera distance to max (5) when playing as a titan, you won't even appear on your minimap.

**To Reproduce**
1. Go to any map and spawn in (more noticeable as a titan)
2. Look at your minimap
3. Change your camera distance to 5
4. Look at the minimap again and notice your icon is no longer centered (or has become so off-centered it no longer appears on the minimap)

**Expected behavior**
The minimap should be centered based on the player character's location instead of his camera's location, so regardless of your camera distance setting your indicator will always be at the center of your minimap.

**Screenshots**
My indicator as a human on 5 camera distance: 
Screenshot (376)

My indicator as a titan on 1 camera distance:
Screenshot (377)

**Contributor instructions**
1. Try to recreate the bug via the reported reproduction steps. If the bug can be reproduced, remove the investigation label. If the bug cannot be reproduced, please add a comment, attach "On Hold" label and inform the submitter that the bug cannot be reproduced. If there is no reply after 2 weeks, the issue can be closed.
2. Follow the default development team procedures (Create a new branch, apply your fix, create a pull request, submit new builds for testing team in dev squad)
3. Once testing team has confirmed the issue is resolved and the pull request is accepted, the issue can be closed.

### LINKED ISSUE #501: [BUG] Minimap no longer displaying the ground after reaching a certain height
**Describe the bug**
Possibly related to #429, or could at least be fixed at the same time. Basically, when the minimap adjusts its display of the map, it factors in the player/camera height; however, if you go too high on some maps, the minimaps display of the map un-renders the map and turns entirely black.

**To Reproduce**
1. Go to City (probably the easiest map to do it on)
2. Launch yourself to the skybox
3. Look at the minimap as you reach it and fall back down

**Expected behavior**
The minimap should correctly render the map regardless of height.

**Screenshots**
Screenshot (512)

**Contributor instructions**
1. Try to recreate the bug via the reported reproduction steps. If the bug can be reproduced, remove the investigation label. If the bug cannot be reproduced, please add a comment, attach "On Hold" label and inform the submitter that the bug cannot be reproduced. If there is no reply after 2 weeks, the issue can be closed.
2. Follow the default development team procedures (Create a new branch, apply your fix, create a pull request, submit new builds for testing team in dev squad)
3. Once testing team has confirmed the issue is resolved and the pull request is accepted, the issue can be closed.

### LINKED ISSUE #503: [BUG] The markers for Titans on the minimap don't update when the Titan is killed
**Describe the bug**
This is a minor bug, as it exists in AoTTG1 as well, but on the minimap, the indicator for a Titan's location (the yellow circle) does not disappear immediately upon the titan dying, instead disappearing only once the Titan's model has disappeared. Only reason I'm reporting it is since the counter for remaining titans now updates immediately upon the titan dying rather than despawning, it may be worth making the minimap match for the sake of consistency.

**To Reproduce**
1. Go to any map/gamemode with Titans
2. Spawn in as a human and kill one
3. Look at the minimap and notice when the yellow dot indicating the Titan's position disappears

**Expected behavior**
If wanting to match the kill feed at the top of the screen, the yellow dot indicator should disappear at the same time as the Titan is killed rather than unloaded from the level

**Contributor instructions**
1. Try to recreate the bug via the reported reproduction steps. If the bug can be reproduced, remove the investigation label. If the bug cannot be reproduced, please add a comment, attach "On Hold" label and inform the submitter that the bug cannot be reproduced. If there is no reply after 2 weeks, the issue can be closed.
2. Follow the default development team procedures (Create a new branch, apply your fix, create a pull request, submit new builds for testing team in dev squad)
3. Once testing team has confirmed the issue is resolved and the pull request is accepted, the issue can be closed.

### LINKED ISSUE #511: [BUG] Minimap updates oddly when the player is moving in certain ways
**Describe the bug**
The player indicator icon on the minimap updates oddly when the player is traveling at certain speeds/angles. This is especially noticeable when the player is travelling up/down at certain angles.

**To Reproduce**
1. Go to City or Forest
2. Swing around a pillar or tree with a fair amount of speed
3. Looking at the minimap and watch the yellow triangle indicating the player's location
4. Stand beneath something tall and hook yourself upwards
5. Look at the minimap as you travel upwards and back down

**Expected behavior**
The icon indicating the player's location on the minimap should update smoothly and not teleporting or jump around oddly.

**Video**
Showcasing horizontal movement: https://youtu.be/9r9BCEIXdDA
Showcasing vertical movement: https://youtu.be/g1X6d-z1bQw

**Contributor instructions**
1. Try to recreate the bug via the reported reproduction steps. If the bug can be reproduced, remove the investigation label. If the bug cannot be reproduced, please add a comment, attach "On Hold" label and inform the submitter that the bug cannot be reproduced. If there is no reply after 2 weeks, the issue can be closed.
2. Follow the default development team procedures (Create a new branch, apply your fix, create a pull request, submit new builds for testing team in dev squad)
3. Once testing team has confirmed the issue is resolved and the pull request is accepted, the issue can be closed.

### PR DESCRIPTION
**Pull request by DaSquyd**
Fixes for #429 #501 #503 #511

### CODE REVIEW

**Comment by atirutw** [comment]
Hey @DaSquyd, mind if I request some stuff? I would like you to bump up the camera height from the player, As of now, it's still having problems with things that are far above the player(for example, the walls). The camera height would at least have to be able to cover the wall's height.

**Comment by DaSquyd** [comment]
@WattanaGaming The camera already does cover the wall height in this PR. It's at a static Y value of (I think) 250 Unity units. I asked for the highest hill size in #533 because if it's higher than 250, it'll clip like it does in the picture provided there. I can bump that number up pretty easily and quickly, but make sure that mountains can't go above 250 first.

**Comment by atirutw** [comment]
Let's continue the discussion in Discord 🤔

File: Assets/MinimapIcon.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿using UnityEngine;
+
+namespace AOT.UI
```
Comment (Lithrun) [inline Assets/MinimapIcon.cs]: Namespace should match based on the location of the class. Additionally, don't place them inside the root folder

File: Assets/MinimapIcon.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+    public class MinimapIcon : MonoBehaviour
+    {
+        private Rigidbody rb;
```
Comment (Lithrun) [inline Assets/MinimapIcon.cs]: Avoid using abbreviations

File: Assets/MinimapIcon.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        private readonly float speed = 1000f;
+
+        private void Start()
+        {
+            rb = GetComponentInParent<Rigidbody>();
+        }
+
+        private void LateUpdate()
+        {
+            if (transform != null)
```
Comment (Lithrun) [inline Assets/MinimapIcon.cs]: Since this is a MonoBehavior, is it even possible for a transform to be null?

**Review by Lithrun — CHANGES_REQUESTED** [review]

**Review by Lithrun — APPROVED** [review]
All good now. Should be able to get merged after the CI is done

**Pull request merged**